salt/states/saltmod.py

```diff
@@ -689,6 +732,137 @@ def runner(name, **kwargs):
     return ret
 
 
+def parallel_runners(name, runners):
+    '''
+    Executes multiple runner modules on the master in parallel.
+
+    .. versionadded:: 2017.x.0 (Nitrogen)
+
+    A separate thread is spawned for each runner. This state is intended to be
+    used with the orchestrate runner in place of the ``saltmod.runner`` state
+    when different tasks should be run in parallel. In general, Salt states are
+    not safe when used concurrently, so ensure that they are used in a safe way
+    (e.g. by only targeting separate minions in parallel tasks).
+
+    name:
+        name identifying this state. The name is provided as part of the
+        output, but not used for anything else.
+
+    runners:
+        list of runners that should be run in parallel. Each element of the
+        list has to be a dictionary. This dictionary's name entry stores the
+        name of the runner function that shall be invoked. The optional kwarg
+        entry stores a dictionary of named arguments that are passed to the
+        runner function.
+
+    .. code-block:: yaml
+
+        parallel-state:
+           saltext.parallel-runner:
+             - runners:
+               - name: state.orchestrate
+                 kwarg:
+                   mods: orchestrate_state_1
```

Comment:
The `[{"name": "...", "kwarg": {...}]` makes _perfect_ sense as an easy-to-write/easy-to-use data structure...but it's not canonical Salt. I can't think of another place in Salt that has something similar so I think it's worth avoiding solely for that reason. Other suggestions welcome, but the following suggestion matches syntaxes found elsewhere in Salt. You can use [repack_dictlist](https://github.com/saltstack/salt/blob/71410047f0fd6126ccd59e37ff4724bf1a0f2d88/salt/utils/__init__.py#L2678) to convert the list of dicts to a dict.

```
             - runners:
               - name: state.orchestrate
               - kwarg:
                    mods: orchestrate_state_1
```